src/details/ArborX_NeighborList.hpp

```diff
@@ -0,0 +1,138 @@
+
+  using Details::HalfTraversal;
+
+  using MemorySpace =
+      typename AccessTraits<Primitives, PrimitivesTag>::memory_space;
+  BVH<MemorySpace> bvh(space, primitives);
+  int const n = bvh.size();
+
+  Kokkos::Profiling::pushRegion(
+      "ArborX::Experimental::HalfNeighborList::Count");
+
+  KokkosExt::reallocWithoutInitializing(space, offsets, n + 1);
+  Kokkos::deep_copy(space, offsets, 0);
+  HalfTraversal(
+      space, bvh, KOKKOS_LAMBDA(int i, int) { ++offsets(i); },
+      NeighborListPredicateGetter{radius});
+  exclusivePrefixSum(space, offsets);
+  KokkosExt::reallocWithoutInitializing(space, indices,
+                                        KokkosExt::lastElement(space, offsets));
+
+  Kokkos::Profiling::popRegion();
+  Kokkos::Profiling::pushRegion("ArborX::Experimental::HalfNeighborList::Fill");
+
+  auto counts =
+      KokkosExt::clone(space, Kokkos::subview(offsets, std::make_pair(0, n)),
+                       "ArborX::Experimental::HalfNeighborList::counts");
+  HalfTraversal(
+      space, bvh, KOKKOS_LAMBDA(int i, int j) { indices(counts(i)++) = j; },
+      NeighborListPredicateGetter{radius});
+  Kokkos::Profiling::popRegion();
+
+  Kokkos::Profiling::popRegion();
+}
+
+template <class ExecutionSpace, class Primitives, class Offsets, class Indices>
+void findFullNeighborList(ExecutionSpace const &space,
+                          Primitives const &primitives, float radius,
+                          Offsets &offsets, Indices &indices)
+{
+  Kokkos::Profiling::pushRegion("ArborX::Experimental::FullNeighborList");
+
+  using Details::HalfTraversal;
+
+  using MemorySpace =
+      typename AccessTraits<Primitives, PrimitivesTag>::memory_space;
+  BVH<MemorySpace> bvh(space, primitives);
+  int const n = bvh.size();
+
+  Kokkos::Profiling::pushRegion(
+      "ArborX::Experimental::FullNeighborList::Count");
+
+  KokkosExt::reallocWithoutInitializing(space, offsets, n + 1);
+  Kokkos::deep_copy(space, offsets, 0);
+  HalfTraversal(
+      space, bvh,
+      KOKKOS_LAMBDA(int i, int j) {
+        Kokkos::atomic_increment(&offsets(i));
+        Kokkos::atomic_increment(&offsets(j));
+      },
+      NeighborListPredicateGetter{radius});
+  exclusivePrefixSum(space, offsets);
+  KokkosExt::reallocWithoutInitializing(space, indices,
+                                        KokkosExt::lastElement(space, offsets));
+
+  Kokkos::Profiling::popRegion();
+  Kokkos::Profiling::pushRegion("ArborX::Experimental::FullNeighborList::Fill");
+
+  auto counts =
+      KokkosExt::clone(space, Kokkos::subview(offsets, std::make_pair(0, n)),
+                       "ArborX::Experimental::FullNeighborList::counts");
+  HalfTraversal(
+      space, bvh, KOKKOS_LAMBDA(int i, int j) { indices(counts(i)++) = j; },
+      NeighborListPredicateGetter{radius});
+  Kokkos::Profiling::popRegion();
+
+  auto counts_copy = KokkosExt::clone(space, counts, counts.label() + "_copy");
```

Comment:
Should a `Copy` profiling region to include the clone, I think.

Reply:
Done